src/GafferOSL/ShadingEngine.cpp

```diff
@@ -912,7 +927,62 @@ OSL::ShadingSystem *shadingSystem()
 
 	g_shadingSystem->attribute( "commonspace", "object" );
 
-	return g_shadingSystem;
+	bool requestBatch = true;
+	if( const char *requestBatchVar = getenv( "GAFFER_OSL_USESIMD" ) )
+	{
+		requestBatch = std::string( requestBatchVar ) != "0";
+	}
+
+#ifdef SHADE_BATCHED
+		// If we wanted to request fused-multiply-add, we would request it here with:
+		// `g_shadingSystem->attribute( "llvm_jit_fma", 1 );`
+		// This is a small improvement in performance ( I couldn't detect a change in my test ),
+		// with the downside that the floating point rounding is slightly different.  The
+		// small difference in rounding is actully more likely to be better than to be worse, so
+		// it all sounds pretty good ... except that the slight difference in rounding may vary
+		// between architectures ... this could be pretty bad for a facility with a heterogenous
+		// farm ... getting a subtle flicker because frames that hit out of date farm blades
+		// render ever-so-slightly darker is not much fun.
+		g_shadingSystem->attribute( "llvm_jit_fma", 1 );
+
+		g_shadingSystemBatchSize = 1;
+		if( requestBatch && g_shadingSystem->configure_batch_execution_at( 16 ) )
+		{
+			g_shadingSystemBatchSize = 16;
+		}
+		else if( requestBatch && g_shadingSystem->configure_batch_execution_at( 8 ) )
+		{
+			g_shadingSystemBatchSize = 8;
+		}
+
+		if( requestBatch && g_shadingSystemBatchSize == 1 )
+		{
+			g_shadingSystem = nullptr;
+			throw IECore::Exception( "Cannot initialize OSL for batch shading.  This requires OSL to have been built with -DUSE_BATCHED for an architecture matching your processor.  If you do not require high performance OSL, you may declare the environment variable GAFFER_OSL_USESIMD=0 to use the slow path." );
+		}
+
+		if( !requestBatch )
+		{
+			msg( Msg::Info, "ShadingEngine", "Initialized shading system without batched shading support" );
+		}
+		else
+		{
+			ustring llvm_jit_target;
+			g_shadingSystem->getattribute("llvm_jit_target", llvm_jit_target);
+			int llvm_jit_fma;
+			g_shadingSystem->getattribute("llvm_jit_fma", llvm_jit_fma);
+
+			msg( Msg::Info, "ShadingEngine", boost::format( "Initialized shading system with support for %i-wide batched shading.  Architecture: %s, Fused-Multiply-Add: %s" ) % g_shadingSystemBatchSize % llvm_jit_target.string() % ( llvm_jit_fma ? "Enabled" : "Disabled" ) );
+		}
+#else
+		if( requestBatch )
+		{
+			g_shadingSystem = nullptr;
+			throw IECore::Exception( "Gaffer was not built with support for batched shading in OSL.  Either build Gaffer with version of OSL more recent than 1.12.8.0, or if you do not require high performance OSL, you may declare the environment variable GAFFER_OSL_USESIMD=0 to use the slow path." );
+		}
+#endif
```

Comment:
I think we should omit this. I probably didn't communicate this very well, but my intention was that the environment variable and exception would all be inside an `OSL_USE_BATCHED` block.

Reply:
Addressed in ae15e51